infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -340,62 +354,68 @@ private void connectToClusterManager() {
      * - Start Log Replication Server(listens and processes incoming requests from the Source)
      * Both:
      * - Metadata Managers which maintain metadata related to replication and its status
-     * @param role
+     * @param topology
      */
-    private void performRoleBasedSetup(ClusterRole role) {
-        if (role != ClusterRole.SOURCE && role != ClusterRole.SINK) {
-            log.debug("Cluster role is {}.  Not performing role-based setup.", role);
+    private void performRoleBasedSetups(TopologyDescriptor topology) {
+        if (topology.getRemoteSourceClusters().isEmpty() && topology.getRemoteSinkClusters().isEmpty()) {
+            log.debug("There is no remote-source/remote-sink for this clusters. Not performing role-based setup.");
```

Comment:
The local cluster ID would be printed just a line above this, from #788.
Let me know if still think it's good to have along with the msg because I do see a value as the log msg will be complete in itself.